core/src/apps/common/paths.py

```diff
@@ -372,3 +372,42 @@ def _path_item(i: int) -> str:
 
 def unharden(item: int) -> int:
     return item ^ (item & HARDENED)
+
+
+def get_account_num(address_n: list[int], pattern: str | list[str]) -> int:
+    if isinstance(pattern, str):
+        pattern = [pattern]
+
+    # Trying all possible patterns - at least one should match
+    for patt in pattern:
+        try:
+            return _get_account_num_single(address_n, patt)
+        except ValueError:
+            pass
+
+    # This function should not raise
+    return 0
+
+
+def _get_account_num_single(address_n: list[int], pattern: str) -> int:
+    # Validating address_n is compatible with pattern
+    # (need to have same length and same hardening)
+    address_str = address_n_to_str(address_n)
+    addr_parts = address_str.split("/")
+    patt_parts = pattern.split("/")
+    if len(addr_parts) != len(patt_parts):
+        raise ValueError
+    for add, pat in zip(addr_parts, patt_parts):
+        if add.count("'") != pat.count("'"):
+            raise ValueError
```

Comment:
Why not just call `PathSchema.parse(pattern).match(address_n)` here instead of reimplementing subset of it?

Reply:
Right, done in [b558b39](https://github.com/trezor/trezor-firmware/pull/3299/commits/b558b39a7128223d023b95140d0257bf93e1261f) - I did not like this much as it needs `slip44_id` to be forwarded, and in `ETH` it is not even defined (using `address_n[1]`), but I agree it is better